### PR TITLE
Feature/color variations function

### PR DESCRIPTION
**Pull request by davecarter**
##  PR Summary

This PR addresses [this issue](https://github.com/SUI-Components/theme-basic/issues/53)
Adds a Sass function to generate a color variation.

Please review

### CODE REVIEW

**Comment by davecarter** [comment]
ups! sorry 😅

File: src/utils/_colors.scss [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+// Usage:
+// $c-primary-light: color-variation($c-primary, 1) -> 1 step lighter
+// $c-primary-lighter: color-variation($c-primary, 2) -> 2 steps lighter
+// $c-primary-dark: color-variation($c-primary, -2) -> 2 steps darker
+// $c-primary-darkest: color-variation($c-primary, -4) -> 4 steps darker
+
+@function color-variation($color, $step: 0) {
+  $step-variations-positive: 25, 50, 75, 90;
+  $step-variations-negative: 20, 35, 55, 75;
```
Comment (ruben-martin-lozano) [inline src/utils/_colors.scss]: Le'ts customize this steps outside the mixin in order to let another projects to define its steps and step-values.

File: src/utils/_colors.scss [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  @if $step == 0 {
+    @return $color;
+  }
+
+  @if $step >= 5 {
```
Comment (ruben-martin-lozano) [inline src/utils/_colors.scss]: Then here use the positive.length

File: src/utils/_colors.scss [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    @return mix($c-white, $color, nth($step-variations-positive, 4));
+  }
+
+  @if $step <= -5 {
```
Comment (ruben-martin-lozano) [inline src/utils/_colors.scss]: And here nevative.length

File: src/utils/_colors.scss [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+@function color-variation($color, $step: 0) {
+  $step-variations-positive: 25, 50, 75, 90;
+  $step-variations-negative: 20, 35, 55, 75;
+
+  @if $step == 0 {
+    @return $color;
+  }
+
+  @if $step >= 5 {
+    @return mix($c-white, $color, nth($step-variations-positive, 4));
```
Comment (ruben-martin-lozano) [inline src/utils/_colors.scss]: Instead 4, last item in array

File: src/utils/_colors.scss [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+  }
+
+  @if $step <= -5 {
+    @return mix($c-black, $color, nth($step-variations-negative, 4));
```
Comment (ruben-martin-lozano) [inline src/utils/_colors.scss]: Instead 4 last item in negative array

File: src/utils/_colors.scss (hunk 2 above)
Comment (naxhh) [inline src/utils/_colors.scss]: shouldn't an invalid step just fail?

we can be silently allowing a bug

File: src/utils/_colors.scss (hunk 1 above)
Comment (naxhh) [inline src/utils/_colors.scss]: having different % on the positive than the negative sound kind of coupled to a specific use case?

**Comment by davecarter** [comment]
Hi all! - I've modified this mixin to be more flexible in terms of steps so instead of setting a previously defined array of steps to return defined style guide values only now it has the ability to get an array of steps **but** the implementation differs quite a bit:

To get the 4th darker color we have to:
**Before**:
`color-variation($c-primary, -4);`

**With open approach**:
color-variation($c-primary, nth($step-variations-negative, 4));

In my opinion I'd rather keep the number of steps pre-fixed internally according to style-guide to simplify implementation. What do you think? @SUI-Components/developers

File: src/utils/_colors.scss (hunk 1 above)
Comment (kikoruiz) [inline src/utils/_colors.scss]: It seems that having different variables for each step, variations will fit better for each site/requirement.

**Comment by kikoruiz** [comment]
@davecarter Alright! I see color steps are "common" regarding this: https://zpl.io/18FDdJ

**Comment by davecarter** [comment]
Yes @kikoruiz - That's the reason why they are **inside** the mixin definition

**Comment by ruben-martin-lozano** [comment]
IMO, if this mixin could be used with another style guide from other companies (outside Schibsted) then both arrays should be as parameters (or even better, variables outside the mixin). If not, then is fine.

**Pull request merged**